src/library/coverartdelegate.cpp

```diff
@@ -87,6 +87,15 @@ void CoverArtDelegate::slotCoverFound(const QObject* pRequestor,
 void CoverArtDelegate::paintItem(QPainter *painter,
                              const QStyleOptionViewItem &option,
                              const QModelIndex &index) const {
+    // If the row is not selected, paint the desired background color before
+    // painting the delegate item
+    if (!option.showDecorationSelected || !(option.state & QStyle::State_Selected)) {
+        QVariant bgValue = index.data(Qt::BackgroundRole);
+        if (bgValue.isValid()) {
+            painter->fillRect(option.rect, qvariant_cast<QBrush>(bgValue));
```

Comment:
```
DEBUG_ASSERT(bgValue.canConvert<QBrush>());
```

Reply:
This background painting code is duplicated multiple times in different classes. Please introduce an abstraction to avoid this, a free function should be sufficient.

Reply:
At first I added this once to `TrackItemDelegate::paint`, but for some reason this caused the bg of the BpmDelegate to be painted twice.